data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java

```diff
@@ -77,6 +94,69 @@ public void start() {
                     ctx.status(SC_NOT_FOUND).result(JsonProvider.objectToJSON("Not found"));
                   }
                 }));
+
+    app.start();
+  }
+
+  private static OpenApiOptions getOpenApiOptions() {
+    Info applicationInfo =
+        new Info()
+            .title(StringUtils.capitalize(VersionProvider.CLIENT_IDENTITY))
+            .version(VersionProvider.IMPLEMENTATION_VERSION)
+            .description(
+                "A minimal API specification for the beacon node, which enables a validator "
+                    + "to connect and perform its obligations on the Ethereum 2.0 phase 0 beacon chain.")
+            .license(
+                new License()
+                    .name("Apache 2.0")
+                    .url("https://www.apache.org/licenses/LICENSE-2.0.html"));
+    OpenApiOptions options =
+        new OpenApiOptions(applicationInfo)
+            .path("/swagger-docs")
+            .swagger(new SwaggerOptions("/swagger-ui"));
+    // TODO: allow swagger-ui to be turned off - ideally still leave swagger-docs, just dont add the
+    // swagger-ui endpoint
+    return options;
+  }
+
+  private void addNodeHandlers(ChainStorageClient chainStorageClient) {
+    app.get(GenesisTimeHandler.ROUTE, new GenesisTimeHandler(chainStorageClient));
+    app.get(VersionHandler.ROUTE, new VersionHandler());
+    /*
+     * TODO:
+     *    /node/syncing
+     *  Optional:
+     *    /node/fork
+     */
+  }
+
+  private void addBeaconHandlers(
+      ChainStorageClient chainStorageClient, HistoricalChainData historicalChainData) {
+    // TODO: not in Minimal or optional specified set - some are similar to lighthouse
+    // implementation
+    handlers.add(new BeaconHeadHandler(chainStorageClient));
+    handlers.add(new BeaconChainHeadHandler(chainStorageClient));
+    handlers.add(new BeaconStateHandler(chainStorageClient));
+    handlers.add(new FinalizedCheckpointHandler(chainStorageClient));
+    handlers.add(new BeaconBlockHandler(chainStorageClient, historicalChainData));
```

Comment:
nit: any reason for this ordering? why aren't all the beacon ones together

Reply:
these are all slated for deletion by the time we're done... I will re-order them if you prefer, but they're on borrowed time.